firestore/src/main/firebase_metadata_provider_desktop.cc

```diff
@@ -23,10 +23,12 @@ namespace firebase {
 namespace firestore {
 
 FirebaseMetadataProviderCpp::FirebaseMetadataProviderCpp(const App& app)
-    : gmp_app_id_(app.options().app_id()) {}
+    : heartbeat_controller_(app->GetHeartbeatController()),
```

Comment:
nit: use `Move()` to avoid triggering copy constructor for the SharedPtr. That is:

```
    : heartbeat_controller_(Move(app->GetHeartbeatController())),
```

Very minor improvement but just for good habit.